File: JobSearchProject/ClientApp/src/app/models/babysitterVacancy.ts
import { Vacancy } from "./vacancy";
import { LanguageType } from "./enum";
import { Education } from "./education";

export class BabysitterVacancy extends Vacancy {
    public childNumber: number = null;
    public nativeLanguage: LanguageType = null;
    public otherLanguages: string = null;
    public driverLicense: boolean = null;
    public ownChildren: boolean = null;
    public officialEmployment: boolean = null;
    public medicineBook: boolean = null;
    public specialChild: boolean = null;
    public videoSurveillance: boolean = null;
    public foreignPassport: boolean = null;
    public travelWithFamily: boolean = null;
    public workingHours: string = null;
    public responsibilities: string = null;
    public criminalRecord: boolean = null;
    public beginningOfWork: string = null;
    public details: string = null;
    public pet: boolean = null;
    public education: Education = new Education();

}

File: JobSearchProject/Models/BabysitterVacancy.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace JobSearchProject.Models
{
    public class BabysitterVacancy : Vacancy
    {
        [Required]
        public int ChildNumber { get; set; }

        [Required]
        public LanguageType NativeLanguage { get; set; }

        public string OtherLanguages { get; set; }

        public bool DriverLicense { get; set; }

        public bool OwnChildren { get; set; }

        public bool OfficialEmployment { get; set; }

        [Required]
        public bool MedicineBook { get; set; }

        public bool SpecialChild { get; set; }

        public bool VideoSurveillance { get; set; }

        public bool ForeignPassport { get; set; }

        public bool TravelWithFamily { get; set; }

        public string WorkingHours { get; set; }

        public string Responsibilities { get; set; }

        public bool CriminalRecord { get; set; }

        public string BeginningOfWork { get; set; }

        public string Details { get; set; }

        public bool Pet { get; set; }

        public int EducationId { get; set; }

        [ForeignKey("EducationId")]
        public virtual Education Education { get; set; }
    }
}
